app/src/main/java/com/swent/assos/ui/screens/assoDetails/AssoDetails.kt

```diff
@@ -289,8 +292,10 @@ fun TopAssoBar(asso: Association, navigationActions: NavigationActions, viewMode
               onClick = {
                 if (associationFollowed.value) {
                   viewModel.unfollowAssociation(asso.id)
+                  profileViewModel.updateNeeded()
                 } else {
                   viewModel.followAssociation(asso.id)
+                  profileViewModel.updateNeeded()
```

Comment:
You can put the line : profileViewModel.updateNeeded() outside the if-else statement. To avoid code repetition try placing it on line 300.